source/framework/core/src/TRestAnalysisTree.cxx

```diff
@@ -22,36 +22,44 @@
 
 //////////////////////////////////////////////////////////////////////////
 ///
-/// TRestAnalysisTree is TTree but with **managed objects** for the branches to fill.
-/// There are six fixed branches of event information in TRestAnalysisTree: runOrigin,
-/// subRunOrigin, eventID, subEventID, subEventTag and timeStamp. They are pointing
-/// to the corresponding class members inside TRestAnalysisTree. Those branches are
-/// called `event branches`. Additional branches could be added by the user, they will
-/// point to some objects whose addresses are also stored in this class. Those objects
-/// are called `observables`.
+/// TRestAnalysisTree is a TTree but with **managed objects** for the branches to be
+/// filled. Inside a TRestAnalysisTree we find always the following six branches
+/// containing event information: runOrigin, subRunOrigin, eventID, subEventID,
+/// subEventTag and timeStamp. Those branches point to the corresponding class
+/// members inside TRestAnalysisTree, we name those branches the `event branches`.
+/// Additional branches can be added by the user, they will point to some objects
+/// whose addresses are also stored in this class. Those objects are called
+/// `observables`.
 ///
-/// In traditional TTree case, the user defines multiple global variables, and add
-/// branches with the address of these variables to the tree. Then the user changes the
-/// value of those variables somewhere in the code, and calls `TTree::Fill()` to create
-/// and save a new entry in the data list.
+/// In the traditional `TTree` case, the user defines multiple global variables,
+/// and adds branches with the address of these variables to the tree. Then the
+/// user changes the value of those variables somewhere in the code, and calls
+/// `TTree::Fill()` to create and save a new entry inside the tree.
 ///
-/// In TRestAnalysisTree, the concept of "Branch" is weakened. We can directly call
-/// `SetObservableValue()` and then `TRestAnalysisTree::Fill()` to do the data saving.
-/// The code could be simplified while sacrificing a little performance. We can use
+/// In TRestAnalysisTree, the concept of "Branch" is weakened. We update the
+/// variables by invoking the `SetObservableValue()` method and then
+/// TRestAnalysisTree::Fill() to generate a new entry inside the tree.
+/// The code will be simplified while sacrificing a little performance. We can use
 /// temporary variable to set observable value directly. We can focus on the analysis
```

Comment:
I do not catch the meaning of these 2 last sentences.

In one hand what we do is to encapsulate the use of the tree through the method `SetObservableValue` so that the user does not need to be playing around with variable generation and branch address creation. Perhaps is something like that we should say here?

Reply:
> what we do is to encapsulate the use of the tree through the method SetObservableValue so that the user does not need to be playing around with direct variable generation and branch address creation

Exactly. That shall be one of the selling point of `TRestAnalysisTree`. We can clarify it in the text.